internal/controller/bootstrap/controlplane_bootstrap_controller.go

```diff
@@ -154,6 +154,18 @@ func (c *ControlPlaneController) Reconcile(ctx context.Context, req ctrl.Request
 	)
 
 	if config.Spec.K0s != nil {
+		storage, found, err := unstructured.NestedMap(config.Spec.K0s.Object, "spec", "storage", "etcd")
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error retrieving storage.etcd: %w", err)
+
+		}
+		if found && storage != nil {
```

Comment:
We probably need to set it even if it's not found in the original config

Reply:
yes, I added a check to set it always if kind storage is not set